service/worker/archiver/util.go

```diff
@@ -96,6 +97,17 @@ func shouldRun(probability float64) bool {
 	return rand.Intn(int(1.0/probability)) == 0
 }
 
+func isHistoryMutated(historyBlob *HistoryBlob, request *ArchiveRequest) bool {
+	lastFailoverVersion := common.Int64Default(historyBlob.Header.LastFailoverVersion)
+	if lastFailoverVersion > request.CloseFailoverVersion {
+		return true
+	}
+
+	isLast := common.BoolDefault(historyBlob.Header.IsLast)
```

Comment:
The return statement is a little long to reason about. Can we short circut here and say 
```
if !isLast {
    return false
}
```